### DISCUSSION TITLE
Modify color of labels actor (i.e. given by `add_point_labels`)

### DISCUSSION THREAD

**Discussion by kerim371** [pyvista/pyvista#6702] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import numpy as np
import pyvista as pv
pl = pv.Plotter()
points = np.array(
    [[0.0, 0.0, 0.0], [1.0, 1.0, 0.0], [2.0, 0.0, 0.0]]
)
labels = ['Point A', 'Point B', 'Point C']
actor = pl.add_point_labels(
    points,
    labels,
    italic=True,
    font_size=20,
    point_color='red',
    point_size=20,
    render_points_as_spheres=True,
    always_visible=True,
    shadow=True,
)

# ============================================================
# Need to modify text color after label actor has been created
# it can't be set as this: `actor.prop.color = "red"` as 
# it doesn' have `actor.prop` property 
# ============================================================

pl.camera_position = 'xy'
pl.show()
```

### Description

Hi,

Can't understand how to edit color of label actor.

I'e succesively changed the color of points actor but label actor doesn't have `actor.prop.` property...

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sun Oct 13 03:24:47 2024 Russia TZ 2 Standard Time

                  OS : Windows (10 10.0.22631 SP0 Multiprocessor Free)
              CPU(s) : 16
             Machine : AMD64
        Architecture : 64bit
                 RAM : 31.4 GiB
         Environment : Python
          GPU Vendor : NVIDIA Corporation
        GPU Renderer : NVIDIA GeForce RTX 3050 Laptop GPU/PCIe/SSE2
         GPU Version : 4.5.0 NVIDIA 546.80
    MathText Support : False

  Python 3.9.11 (tags/v3.9.11:2de452f, Mar 16 2022, 14:33:45) [MSC v.1929 64
  bit (AMD64)]

             pyvista : 0.44.1
                 vtk : 9.3.1
               numpy : 2.0.2
          matplotlib : 3.9.2
              scooby : 0.10.0
               pooch : 1.8.2
              pillow : 10.4.0
             imageio : 2.35.1
             IPython : 8.18.1
            colorcet : 3.1.0
             cmocean : 4.0.3
          ipywidgets : 8.1.5
               scipy : 1.13.1
              meshio : 5.3.5
               trame : 3.6.3
        trame_client : 3.2.5
        trame_server : 3.1.0
           trame_vtk : 2.8.10
       trame_vuetify : 2.7.0
jupyter_server_proxy : 4.4.0
        nest_asyncio : 1.6.0
--------------------------------------------------------------------------------
```

**Comment by banesullivan** [comment]
The issue here is that add_point_labels makes two actors, not just the one that is returned. We ought to implement a mechanism to return references to both actors. See https://github.com/pyvista/pyvista/issues/5235

For now, you can hackily retrieve the other actor by specifying the `name` of the actor and using that name to retrieve the actor

```py
import numpy as np
import pyvista as pv
pl = pv.Plotter()
points = np.array(
    [[0.0, 0.0, 0.0], [1.0, 1.0, 0.0], [2.0, 0.0, 0.0]]
)
actor_name = 'my-lables'
labels = ['Point A', 'Point B', 'Point C']
labels_actor = pl.add_point_labels(
    points,
    labels,
    italic=True,
    font_size=20,
    point_color='red',
    point_size=20,
    render_points_as_spheres=True,
    always_visible=True,
    shadow=True,
    name=actor_name
)
# Use the name of the actor and the `-points` suffix that `add_point_labels` adds
points_actor = pl.actors[f"{actor_name}-points"]
points_actor.prop.color = 'black'

pl.camera_position = 'xy'
pl.show()
```

**Reply by kerim371** [reply]
@banesullivan hi,

Thank you for the hint.

I just tried the code you provided and there is a mistype. 
Actually `add_point_labels` seems to return `...-points` actor and to get `labels` actor we need:
```python
labels_actor = pl.actors[f"{actor_name}-labels"]
```

But `labels_actor ` is a `vtkActor2D` which doesn't have `labels_actor.prop` property.
Basically this question becomes more pure vtk question: how to change text color (and probbly background) of `vtkActor2D` that contains text?
```python
c = pv.Color(state.well_color)
labels_actor .GetProperty().SetColor(c.int_rgb)
```
doesn't work by the way.
It seems `PyVista` doesn't use `vtkCaptionActor2D`.

**Reply by banesullivan** [reply]
I'm so sorry! I hastily read your post and thought you wanted to change the points color. We ought to be able to change the label color. I'll look into this

**Reply by kerim371** [reply]
Thank you! Good :)

**Reply by banesullivan** [reply, marked as answer]
It looks like the shape color is controlled on the 2D actor's mapper:

https://github.com/pyvista/pyvista/blob/4b905fd9899b0b5113bd5f00b5f1c0377f048a3d/pyvista/plotting/plotter.py#L5700

And the text colors are controlled on the `vtkPointSetToLabelHierarchy`

https://github.com/pyvista/pyvista/blob/4b905fd9899b0b5113bd5f00b5f1c0377f048a3d/pyvista/plotting/plotter.py#L5716

We can retrieve these VTK components from the label actor:

```py
mapper = labels_actor.GetMapper()
mapper.SetBackgroundColor(pv.Color('blue').float_rgb)

text_prop = mapper.GetInputAlgorithm().GetTextProperty()
text_prop.SetColor(pv.Color('red').float_rgb)

pl.render()
```

**Reply by kerim371** [reply]
@banesullivan thank you very much!
That works perfect!
